app/actions/remote/entry/login.ts

```diff
@@ -90,89 +79,22 @@ export async function loginEntry({serverUrl, user, deviceToken}: AfterLoginArgs)
-
-            if (initialTeam) {
-                const rolesToFetch = new Set<string>([...user.roles.split(' '), ...teamRoles]);
-
-                // fetch channels / channel membership for initial team
-                chData = await fetchMyChannelsForTeam(serverUrl, initialTeam.id, false, 0, true);
-                if (chData.channels?.length && chData.memberships?.length) {
-                    const {channels, memberships} = chData;
-                    const channelIds = new Set(channels?.map((c) => c.id));
-                    for (let i = 0; i < memberships!.length; i++) {
-                        const member = memberships[i];
-                        if (channelIds.has(member.channel_id)) {
-                            member.roles.split(' ').forEach(rolesToFetch.add, rolesToFetch);
-                        }
-                    }
-
-                    // fetch user roles
-                    rData = await fetchRolesIfNeeded(serverUrl, Array.from(rolesToFetch), true);
-
-                    // select initial channel only on Tablets
-                    if (isTabletDevice) {
-                        initialChannel = selectDefaultChannelForTeam(channels!, memberships!, initialTeam!.id, rData.roles, user.locale);
-                    }
-                }
-            }
-        }
-
-        const modelPromises = await prepareModels({operator, teamData, chData, prefData, initialTeamId: initialTeam?.id});
-
-        const systemModels = prepareCommonSystemValues(
-            operator,
-            {
-                config: clData.config || ({} as ClientConfig),
-                license: clData.license || ({} as ClientLicense),
-                currentTeamId: initialTeam?.id || '',
-                currentChannelId: initialChannel?.id || '',
-            },
-        );
-        if (systemModels) {
-            modelPromises.push(systemModels);
-        }
-
-        if (initialTeam) {
-            const th = addTeamToTeamHistory(operator, initialTeam.id, true);
-            modelPromises.push(th);
-        }
-
-        if (initialTeam && initialChannel) {
-            try {
-                const tch = addChannelToTeamHistory(operator, initialTeam.id, initialChannel.id, true);
-                modelPromises.push(tch);
-            } catch {
-                // do nothing
-            }
-        }
-
-        if (rData?.roles?.length) {
-            const roles = operator.handleRole({roles: rData.roles, prepareRecordsOnly: true});
-            modelPromises.push(roles);
+        let switchToChannel = false;
+        const {models, initialTeamId, initialChannelId, prefData, teamData, chData} = entryData;
+        if (initialChannelId && isTabletDevice) {
+            switchToChannel = true;
+            switchToChannelById(serverUrl, initialChannelId, initialTeamId);
+        } else {
+            setCurrentTeamAndChannelId(operator, initialTeamId, initialChannelId);
```

Comment:
initial channel should only be set for tablets, right? probably the login entry missed this as it was the first thing added to the code, basically.

```suggestion
            setCurrentTeamAndChannelId(operator, initialTeamId, '');
```